go/vt/vtgate/planbuilder/route.go

```diff
@@ -146,10 +146,18 @@ func (rb *route) SetRHS() {
 // the expression contains a non-pushable subquery. ajoin can be nil
 // if the join is on a ',' operator.
 func (rb *route) merge(rhs *route, ajoin *sqlparser.JoinTableExpr) (builder, error) {
+	sel, ok := rb.Select.(*sqlparser.Select)
+	if !ok {
+		return nil, errors.New("unsupported: merge union")
```

Comment:
This (and below) should be dead code. When we do these merges, we're only analyzing `from` clauses. So, the routes should all contain `Select` only (not `Union`). It's ok to optimistically assert like mentioned earlier.